scriptmodules/emulators/mupen64plus/mupen64plus.sh

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env bash
+VIDEO_PLUGIN="$1"
+ROM="$2"
+configdir="/opt/retropie/configs"
+
+# arg 1: device (/dev/input/js*)
+function getDeviceName() {
+    local device="$1"
+    local device_name=""
+
+    # read device name
+    device_name=`cat /sys/class/input/${device}/device/name`
+
+    echo "\"$device_name\""
+    return
+}
```

Comment:
device_name isn't needed since the value is echoed, but I would just get rid of the whole function - only used in one place, and /sys/class/input/${device}/device/name can be used directly (note in bash you can do value=$(<file) instead of cat